internal/encoded.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package internal
+
+import (
+	"reflect"
+
+	"go.uber.org/cadence/encoded"
+)
+
+// defaultDataConverter uses thrift encoder/decoder when possible, for everything else use json.
+type defaultDataConverter struct{}
+
+var defaultJsonDataConverter = &defaultDataConverter{}
+
+// DefaultDataConverter is default data converter used by Cadence worker
+var DefaultDataConverter = getDefaultDataConverter()
```

Comment:
Why do you we need two of these variables?

Reply:
`defaultJsonDataConverter` is to avoid creating struct for each get call. 
`DefaultDataConverter` is just for exposing, since getDefaultDataConverter method is not exposed.

Reply:
I'm confused. Why you just don't change visibility of getDefaultDataConverter?